inference-engine/src/inference_engine/network_serializer.cpp

```diff
@@ -4,165 +4,261 @@
-                        if (childNode->input_value(iport).get_node() == parentPort.get_node()) {
-                            auto itTo = matching.find(childNode->shared_from_this());
-                            if (itTo == matching.end()) {
-                                THROW_IE_EXCEPTION << "Broken edge form layer "
-                                                   << parentNode->get_friendly_name() << " to layer "
-                                                   << childNode->get_friendly_name()
-                                                   << "during serialization of IR";
-                            }
-                            pugi::xml_node edge = edges.append_child("edge");
-                            edge.append_attribute("from-layer").set_value(itFrom->second);
-                            edge.append_attribute("from-port").set_value(oport + parentNode->get_input_size());
-
-                            edge.append_attribute("to-layer").set_value(itTo->second);
-                            edge.append_attribute("to-port").set_value(iport);
-                        }
-                    }
-                }
-            }
-        }
+    for (auto i : node->inputs()) {
+      auto source_output = i.get_source_output();
+      auto source_node = source_output.get_node();
+      auto current_node = i.get_node();
+
+      Edge e{};
+      e.from_layer = layer_ids[source_node];
+      e.from_port = source_node->get_input_size() + source_output.get_index();
+      e.to_layer = layer_ids[current_node];
+      e.to_port = i.get_index();
+      edges.push_back(e);
     }
+  }
+  std::sort(begin(edges), end(edges), [](const Edge &a, const Edge &b) -> bool {
+    return a.from_layer < b.from_layer;
+  });
+  return edges;
 }
 
-}  // namespace
-
-void SerializeV10(const std::string& xmlPath, const std::string& binPath,
-                  const InferenceEngine::ICNNNetwork& network) {
-    if (auto function = network.getFunction()) {
-        // A flag for serializing executable graph information (not complete IR)
-        bool execGraphInfoSerialization = true;
-
-        // go over all operations and check whether performance stat is set
-        for (const auto & op : function->get_ops()) {
-            auto & rtInfo = op->get_rt_info();
-            if (rtInfo.find(ExecGraphInfoSerialization::PERF_COUNTER) == rtInfo.end()) {
-                execGraphInfoSerialization = false;
-                break;
-            }
-        }
+ConstantAtributes dump_constant_data(std::vector<uint8_t> &bin,
+                                     const ngraph::op::Constant &c) {
+  ConstantAtributes attr;
+  const uint8_t *p = reinterpret_cast<const uint8_t *>(c.get_data_ptr());
+  attr.size = ngraph::shape_size(c.get_shape()) * c.get_element_type().size();
+  attr.offset = bin.size();
+  bin.insert(end(bin), p, p + attr.size);
+  return attr;
+}
+
+void ngfunction_2_irv10(pugi::xml_document &doc, std::vector<uint8_t> &bin,
```

Comment:
If we are talking about the task 32765 we can move this to transformation library.

If we move this code to transformation library I propose to add a parameter for transformation which should allow to select the version of IR for serialization (by default 10).
It can help in the future support different IR versions.

Reply:
I've moved code to transformation library in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/1b867e6a29f4ac66e3fce509388d3848b83d4cd3) and added versioning in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/d7c0eb6ae4f6b099c1802388eb908461cfcf7537l). Previously I used some code from inference_engine so I needed to remove this code, because transformation library cannot depend on ie - it would create circular dependency.